packages/block-library/src/template-part/edit/selection/template-part-previews.js

```diff
@@ -178,7 +179,10 @@ function TemplatePartSearchResults( {
 	const currentShownTPs = useAsyncList( filteredTPs );
 
 	return filteredTPs.map( ( templatePart ) => (
-		<PanelGroup key={ templatePart.id } title={ templatePart.meta.theme }>
+		<PanelGroup
+			key={ templatePart.id }
+			title={ templatePart.meta.theme || __( 'Custom' ) }
+		>
```

Comment:
Change makes sense to me, but I've noticed that when searching, template parts aren't grouped by theme, but each part is in its own theme group:

| Normal List | Search Results |
| - | - |
| <img width="327" alt="Screenshot 2020-10-30 at 18 40 00" src="https://user-images.githubusercontent.com/2070010/97744621-7a7d6700-1adf-11eb-9dde-87fbab96bd48.png"> | <img width="333" alt="Screenshot 2020-10-30 at 18 40 12" src="https://user-images.githubusercontent.com/2070010/97744606-718c9580-1adf-11eb-952f-338eb593529e.png"> |

(This is not related to this PR, but I just thought of mentioning it)

Reply:
Yeah, the search results group them individually as they are ordered a bit naively by the priorities:
1. index match is found in the slug.
2. index match is found in the theme.
I think a lot of this does need to be rethought in the future.